connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/DeadLetterQueueReporter.java

```diff
@@ -49,22 +47,14 @@
     private static final short DLQ_MAX_DESIRED_REPLICATION_FACTOR = 3;
     private static final int DLQ_NUM_DESIRED_PARTITIONS = 1;
 
-    public static final String PREFIX = "errors.deadletterqueue";
+    private final SinkConnectorConfig connConfig;
 
-    public static final String DLQ_TOPIC_NAME = "topic.name";
-    public static final String DLQ_TOPIC_NAME_DOC = "The name of the topic where these messages are written to.";
-    public static final String DLQ_TOPIC_DEFAULT = "";
-
-    private DeadLetterQueueReporterConfig config;
     private KafkaProducer<byte[], byte[]> kafkaProducer;
     private ErrorHandlingMetrics errorHandlingMetrics;
 
-    private static final ConfigDef CONFIG_DEF = new ConfigDef()
-            .define(DLQ_TOPIC_NAME, ConfigDef.Type.STRING, DLQ_TOPIC_DEFAULT, ConfigDef.Importance.HIGH, DLQ_TOPIC_NAME_DOC);
-
     public static DeadLetterQueueReporter createAndSetup(WorkerConfig workerConfig,
-                                                         ConnectorConfig connConfig, Map<String, Object> producerProps) {
-        String topic = connConfig.getString(PREFIX + "." + DLQ_TOPIC_NAME);
+                                                         SinkConnectorConfig connConfig, Map<String, Object> producerProps) {
+        String topic = connConfig.dlqTopicName();
 
         try (AdminClient admin = AdminClient.create(workerConfig.originals())) {
             if (!admin.listTopics().names().get().contains(topic)) {
```

Comment:
In the first iteration of this code, we checked the number of brokers available on the cluster before setting the replication factor. But that has apparently caused issues in the past where having fewer brokers during the time this code was called, created a topic with less than desired replication factor. Hence, we decided to simply put this number as it is.

A config would mean changes to the KIP? IIRC, we had it in the first iteration of the KIP, but reviewers wanted a simpler first version.

Reply:
I think there are three options:
1. Leave as is, but improve the error message and suggest users manually create the topic. This has usability implications, though I don't expect too many demos will enable DLQ. This does not require changing the KIP.
2. Add one configuration to set the replication factor. Like the other Connect settings that deal with replication factors, this would be low priority and should default to 3, which should be fine for many production use cases when DLQ is implemented. The error message should still be modified to be useful, but that error should happen less frequently. This means updating the KIP.
3. Never even attempt to create the topic, so fail if it does not exist and improve the error message with more useful detail. This means updating the KIP to at least suggest that the topic will not be created.

IMO, option 2 is the most usable, but it still requires users with small clusters (e.g., a dev environment) that are enabling DLQ to set this. Options 1 and 3 are both less useable.

We should probably create a Jira issue and discuss it there, but it needs to be done quickly and it should be marked a blocker.

Reply:
I had pointed out in the initial review that the heuristic to guess the desired RF seemed of little value. I'm also for reconsidering.

Reply:
@kkonstantine there is no heuristic here, do you mean have the replication factor come in from a config field?